docs/src/IconButton.doc.js

```diff
@@ -1,419 +1,862 @@
-  <Tooltip text="Custom Icon">
-    <IconButton
-      accessibilityLabel="Open edit modal to edit Board"
-      dangerouslySetSvgPath={{ __path: 'M23 5v14a4 4 0 0 1-4 4H5a4 4 0 0 1-4-4v-5.5h10.258l-1.94 1.939a1.5 1.5 0 0 0 2.121 2.122L17 12l-5.561-5.561a1.501 1.501 0 0 0-2.121 2.122l1.94 1.939H1V5a4 4 0 0 1 4-4h14a4 4 0 0 1 4 4'}}
-    />
-  </Tooltip>
-</Flex>
+  <MainSection name="Variants">
+    <MainSection.Subsection
+      title="Role"
+      columns={2}
+      description={`IconButton can be use for navigation and actions. If IconButton acts as a link, set \`role = link\` and pass role-specific [props](#role_linkProps).
+
+
+\`target\` is optional and defines the frame or window to open the anchor defined on href:
+* "null" opens the anchor in the same window.
+* "blank" opens the anchor in a new window.
+* "self" opens an anchor in the same frame.
+
+\`rel\` is optional. Use "follow" for external links to specify to web crawlers not follow the link.
+
+IconButtons that act as links can be paired with [OnLinkNavigationProvider](/OnLinkNavigationProvider). See [OnLinkNavigationProvider](/OnLinkNavigationProvider) to learn more about link navigation.
+
+If IconButton acts as a button, pass role-specific [props](#role_buttonProps).`}
+    >
+      <MainSection.Card
+        cardSize="md"
+        defaultCode={`
+<Tooltip text="Button">
+  <IconButton
+    accessibilityLabel="This IconButton is an example of IconButton acting as a button"
+    icon="share"
+    onClick={() => {}}
+  />
+</Tooltip>
 `}
-  />,
-);
+      />
+      <MainSection.Card
+        cardSize="md"
+        defaultCode={`
+<Tooltip text="Link">
+  <IconButton
+    accessibilityLabel="This IconButton is an example of IconButton acting as a link"
+    icon="visit"
+    role="link"
+    target="blank"
+    href="https://www.pinterest.com"
+  />
+</Tooltip>
+`}
+      />
+    </MainSection.Subsection>
+    <MainSection.Subsection
+      title="Size"
+      description={`IconButton is available in 5 fixed sizes:
 
-card(
-  <Example
-    id="ref"
-    name="ref"
-    defaultCode={`
+1. \`xl\` (56px)
+    Extra large IconButtons should be used sparingly and only in places where the UI has a case for an extra-large IconButton.
+2. \`lg\` (48px)
+    Large is the only size that should be used on Pinner surfaces.
+3. \`md\` (40px)
+    Medium is the size used on more dense UI such as business surfaces or internal tools.
+4. \`sm\` (32px)
+    Small Icon button should be used sparingly and only in places where the UI is very dense.
+5. \`xs\` (24px)
+    Use sparingly and only in places where the UI is very dense or has a case for an extra-small IconButton as they can be hard to see for people with visual impairments.
+
+Use padding sparingly. Padding options are 1-5 representing the padding in boints units. Combine the \`padding\` with \`size\` options for custom icon/button size ratios. If omitted, padding is derived from the default padding for each \`size\` prop.`}
```

Comment:
Use padding sparingly. The padding options are 1-5, which represents the padding in increments of 4 pixels (2 = 8px padding). Combine the \`padding\` with \`size\` options for custom icon/button size ratios. If omitted, padding is derived from the default padding for each \`size\` prop.`}